examples/pod_exec_terminal_size.rs

```diff
@@ -0,0 +1,134 @@
+use futures::{channel::mpsc::Sender, SinkExt, StreamExt, TryStreamExt};
+use k8s_openapi::api::core::v1::Pod;
+use tracing::*;
+
+use kube::{
+    api::{
+        Api, AttachParams, AttachedProcess, DeleteParams, ListParams, PostParams, ResourceExt, TerminalSize,
+        WatchEvent,
+    },
+    Client,
+};
+use tokio::{io::AsyncWriteExt, select};
+
+// check terminal size every 1 second and send it to channel if different
+async fn handle_terminal_size(mut channel: Sender<TerminalSize>) {
+    let (mut width, mut height) = crossterm::terminal::size().unwrap();
+    channel
+        .send(TerminalSize { height, width })
+        .await
+        .expect("fail to write new size to channel");
+
+    loop {
+        tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+        let (n_width, n_height) = crossterm::terminal::size().unwrap();
+        if n_height != height || n_width != width {
+            height = n_height;
+            width = n_width;
+            channel
+                .send(TerminalSize { height, width })
+                .await
+                .expect("fail to write new size to channel");
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+    let client = Client::try_default().await?;
+
+    let pods: Api<Pod> = Api::default_namespaced(client);
+    let p: Pod = serde_json::from_value(serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": { "name": "example" },
+        "spec": {
+            "containers": [{
+                "name": "example",
+                "image": "alpine",
+                // Do nothing
+                "command": ["tail", "-f", "/dev/null"],
+            }],
+        }
+    }))?;
+    // Create pod if don't exist
+    pods.create(&PostParams::default(), &p).await?;
+
+    // Wait until the pod is running, otherwise we get 500 error.
```

Comment:
Let's use `await_condition` to reduce noise.

https://github.com/kube-rs/kube-rs/blob/bf885f05c39c046f68e34f19b95402d3326c1715/examples/pod_portforward.rs#L34-L36